src/cluster_legacy.c

```diff
@@ -63,6 +63,8 @@ void clusterUpdateState(void);
 int clusterNodeCoversSlot(clusterNode *n, int slot);
 list *clusterGetNodesInMyShard(clusterNode *node);
 int clusterNodeAddSlave(clusterNode *master, clusterNode *slave);
+int clusterNodeAddMaster(clusterNode *master);
```

Comment:
I actually don't like this abstraction. I would like it to be conceptually possible for a shard to exist without a master. @PingXie Thoughts?

Reply:
Could you explain a bit more on how such situation can arise?

Reply:
Thank you for taking a look. Will wait until you review the PR.